src/rule-generated-flow-types.js

```diff
@@ -419,57 +447,63 @@ module.exports = {
                 break;
               }
               case 'GenericTypeAnnotation': {
-                const alias = propType.id.name;
-                if (!typeAliasMap[alias]) {
+                const aliasedObjectType = extractReadOnlyType(
+                  resolveTypeAlias(propType, typeAliasMap)
+                );
+                if (!aliasedObjectType) {
                   // The type Alias doesn't exist, is invalid, or is being
                   // imported. Can't do anything.
                   break;
                 }
-                switch (typeAliasMap[alias].type) {
+                switch (aliasedObjectType.type) {
                   case 'ObjectTypeAnnotation': {
                     validateObjectTypeAnnotation(
                       context,
                       Component,
                       importedPropType,
                       propName,
-                      typeAliasMap[alias],
+                      aliasedObjectType,
                       importFixRange
                     );
                     break;
                   }
                   case 'IntersectionTypeAnnotation': {
-                    const objectTypes = typeAliasMap[alias].types
+                    const objectTypes = aliasedObjectType.types
                       .map(intersectedType => {
                         if (intersectedType.type === 'GenericTypeAnnotation') {
-                          return typeAliasMap[intersectedType.id.name];
+                          return extractReadOnlyType(
+                            resolveTypeAlias(intersectedType, typeAliasMap)
+                          );
                         }
                         if (intersectedType.type === 'ObjectTypeAnnotation') {
                           return intersectedType;
                         }
                       })
-                      .filter(Boolean);
+                      .filter(maybeObjectType => {
+                        // GenericTypeAnnotation may not map to an object type
+                        return maybeObjectType &&
+                          maybeObjectType.type === 'ObjectTypeAnnotation';
```

Comment:
The lack of a filter here meant a non-ObjectTypeAnnotation could be passed below